src/router-hooks.tsx

```diff
@@ -110,13 +115,16 @@ export const RouterProvider = ({ children }: { children: ReactNode }) => {
     };
   }, [setState]);
   const navigate = useCallback(
-    (newState: RouterState) => {
+    (newState: RouterState, source?: string) => {
```

Comment:
Please can we type source with e.g. `type NavigationSource = "toolkit-user" | "whatever"` etc.
Otherwise this looks good to me.